core/services/relay/evm/mercury_provider.go

```diff
@@ -100,6 +105,47 @@ func (p *mercuryProvider) ContractTransmitter() ocrtypes.ContractTransmitter {
 	return p.transmitter
 }
 
+func (p *mercuryProvider) ChainReader() relaymercury.ChainReader {
+	return p.chainReader
+}
+
 func (p *mercuryProvider) MercuryServerFetcher() relaymercury.MercuryServerFetcher {
 	return p.transmitter
 }
+
+var _ relaymercury.ChainReader = (*chainReader)(nil)
+
+type chainReader struct {
+	lggr        logger.Logger
+	client      evmclient.Client
+	headTracker httypes.HeadTracker
+}
+
+func NewChainReader(c evmclient.Client, h httypes.HeadTracker, lggr logger.Logger) relaymercury.ChainReader {
+	return &chainReader{
+		client:      c,
+		headTracker: h,
+		lggr:        lggr,
+	}
+}
+
+func (r *chainReader) LatestHead(ctx context.Context) (relaymercury.ChainHead, error) {
+	var err error
+
+	head := r.headTracker.LatestChain()
+	if head == nil {
+		logger.Sugared(r.lggr).AssumptionViolation(
```

Comment:
Let's store the sugared logger in lggr, instead of creating it every time.

Reply:
dropped the logger since the previous PR for multiple block observation removed the fallback to the client as unnecessary.